ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/ReexecutingExecutionPayloadBlockManager.java

```diff
@@ -0,0 +1,172 @@
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.async.Cancellable;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.collections.LimitedSet;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult.FailureReason;
+import tech.pegasys.teku.statetransition.util.FutureItems;
+import tech.pegasys.teku.statetransition.util.PendingPool;
+import tech.pegasys.teku.statetransition.validation.BlockValidator;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class ReexecutingExecutionPayloadBlockManager extends BlockManager {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final int EXPIRES_IN_SLOT = 2;
+  private static final Duration RECHECK_INTERVAL = Duration.ofSeconds(2);
+
+  private final AtomicBoolean reexecutingExecutionPayload = new AtomicBoolean(false);
+  private final Set<SignedBeaconBlock> pendingBlocks = LimitedSet.create(10);
+  private final AsyncRunner asyncRunner;
+
+  private Optional<Cancellable> cancellable = Optional.empty();
+
+  private ReexecutingExecutionPayloadBlockManager(
+      RecentChainData recentChainData,
+      BlockImporter blockImporter,
+      PendingPool<SignedBeaconBlock> pendingBlocks,
+      FutureItems<SignedBeaconBlock> futureBlocks,
+      BlockValidator validator,
+      final AsyncRunner asyncRunner) {
+    super(recentChainData, blockImporter, pendingBlocks, futureBlocks, validator);
+    this.asyncRunner = asyncRunner;
+  }
+
+  public static BlockManager create(
+      final PendingPool<SignedBeaconBlock> pendingBlocks,
+      final FutureItems<SignedBeaconBlock> futureBlocks,
+      final RecentChainData recentChainData,
+      final BlockImporter blockImporter,
+      final BlockValidator validator,
+      final AsyncRunner asyncRunner) {
+    return new ReexecutingExecutionPayloadBlockManager(
+        recentChainData, blockImporter, pendingBlocks, futureBlocks, validator, asyncRunner);
+  }
+
+  @Override
+  public SafeFuture<?> doStart() {
+    return super.doStart()
+        .thenRun(
+            () ->
+                cancellable =
+                    Optional.of(
+                        asyncRunner.runWithFixedDelay(
+                            this::execute,
+                            RECHECK_INTERVAL,
+                            error -> LOG.error("Failed to reexecute execution payloads", error))));
+  }
+
+  @Override
+  protected SafeFuture<?> doStop() {
+    return super.doStop()
+        .thenRun(
+            () -> {
+              cancellable.ifPresent(Cancellable::cancel);
+              cancellable = Optional.empty();
+            });
```

Comment:
I suspect we should cancel first then call super (though I'm probably just being paranoid).